Plugins~/Src/MeshSyncClientBlender/msblenCurveHandler.cpp

```diff
@@ -0,0 +1,119 @@
+#include "pch.h"
+#include "msblenBinder.h"
+#include "msblenCurveHandler.h"
+#include "msblenUtils.h"
+
+namespace bl = blender;
+
+ms::CurvePtr msblenCurveHandler::exportCurve(msblenContextState& state,
+	msblenContextPathProvider& paths,
+	BlenderSyncSettings& settings,
+	const Object* src,
+	MeshSyncClient::AsyncTasksController& asyncTasksController)
+{
+	std::shared_ptr<ms::Curve> ret = ms::Curve::create();
+	
+	Curve* data = (Curve*)src->data;
+
+	ms::Curve& dst = *ret;
+	dst.path = paths.get_path(src);
+	
+	// transform
+	extractTransformData(settings, src, dst);
+
+	if (settings.sync_curves) {
+		auto task = [this, ret, src, data, &state, &settings]() {
+			auto& dst = *ret;
+			doExtractCurveData(state, settings, dst, src, data, dst.world_matrix);
+			state.manager.add(ret);
+		};
+
+		if (settings.multithreaded)
+			asyncTasksController.AddTask(std::launch::async, task);
+		else
+			task();
+	}
+
+	return ret;
+}
+
+void msblenCurveHandler::doExtractCurveData(msblenContextState& state, BlenderSyncSettings& settings, ms::Curve& dst, const Object* obj, Curve* data, mu::float4x4 world)
+{
+	if (settings.sync_curves) {
+		const bool is_editing = data->editnurb != nullptr;
+
+		ListBase nurbs;
+		if (is_editing) {
+			nurbs = data->editnurb->nurbs;
+		}
+		else {
+			nurbs = data->nurb;
+		}
+
+		dst.splines.clear();
+
+		for (auto nurb : blender::list_range((Nurb*)nurbs.first)) {
+			dst.splines.push_back(ms::CurveSpline::create());
+			auto curveSpline = dst.splines.back();
+
+			curveSpline->closed = (nurb->flagu & CU_NURB_CYCLIC) != 0;
+
+			curveSpline->cos.resize_discard(nurb->pntsu);
+			curveSpline->handles_left.resize_discard(nurb->pntsu);
+			curveSpline->handles_right.resize_discard(nurb->pntsu);
+
+			for (int i = 0; i < nurb->pntsu; i++) {
+				BezTriple* bezt = &nurb->bezt[i];
+
+				if (bezt) {
```

Comment:
this can be changed to if(!bezt)continue to reduce nesting

Reply:
done